### PR TITLE
Fix specification for minimum PyTorch version in requirements file

### LINKED ISSUE #15183: pytorch version dependency issue
### Bug description

Please see issue here. 

https://github.com/pypa/pip/issues/11524#issuecomment-1283381523

Is lightning depending on `1.9.*` somewhere as mentioned in the issue there?

### How to reproduce the bug

```python
See https://github.com/pypa/pip/issues/11524#issuecomment-1283381523
```

### Error messages and logs

see https://github.com/pypa/pip/issues/11524#issuecomment-1283381523

### Environment

see https://github.com/pypa/pip/issues/11524#issuecomment-1283381523

### More info

_No response_

cc @Borda

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #15183

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @tchaton @rohitgr7

### CODE REVIEW

**Comment by min-xu-ai** [comment]
Thank you so much @awaelchli. I think it does it!

**Comment by min-xu-ai** [comment]
It would be nice to have it somehow not being introduced again, like a test or a comment. But that's certainly just a nice to have.

**Review by akihironitta — COMMENTED** [review]
If I remember correctly, this was introduced only for our CI to use the latest patch release (1.9.1) within the oldest minor release (1.9). This means that, with this change, it uses 1.9.0 instead of 1.9.1 in `pl-cpu (ubuntu-20.04, 3.7, 1.9, oldest)` job:

```diff
-torch                   1.9.1+cpu
+torch                   1.9.0+cpu
```
- master: https://github.com/Lightning-AI/lightning/actions/runs/3278795963/jobs/5397603420#step:12:383
- this branch: https://github.com/Lightning-AI/lightning/actions/runs/3278757620/jobs/5397513506#step:12:378

---

Hi @min-xu-ai, thank you for your report. Could you check again if the issue is really resolved by this change? I've tried running it locally, but I'm still seeing the conflict.

<details><summary>commands</summary>

Slightly edited https://github.com/pypa/pip/issues/11524.

```console
$ cat requirements.txt
-c req_torch.txt
-i https://pypi.org/simple
torch>=1.9  # or torch>=1.9.*
$ cat req_torch.txt
-i https://download.pytorch.org/whl/torch_stable.html
torch==1.10.2+cu113
$ pip install -r requirements.txt
ERROR: Cannot install torch>=1.9 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested torch>=1.9
    The user requested (constraint) torch==1.10.2+cu113

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
aki@lambda-server1 ~/tmp % pip install -r requirements.txt
```

</details>

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by rohitgr7 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: requirements/lite/base.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #  in case you want to preserve/enforce restrictions on the latest compatible version, add "strict" as an in-line comment
 
 numpy>=1.17.2, <1.23.1
-torch>=1.9.*, <1.13.0
+torch>=1.9, <1.13.0
```
Comment (carmocca) [inline requirements/lite/base.txt:5]: Opening this comment to block the merge until there's confirmation for https://github.com/Lightning-AI/lightning/pull/15184#pullrequestreview-1146920075

**Comment by min-xu-ai** [comment]
OK, I have updated the other pip issue with my testing. It seems that using `1.9.*` also works. 

My new requirements.txt now looks like this and it can install everything correctly:

```
--find-links https://download.pytorch.org/whl/torch_stable.html
torch==1.10.2+cu113

# runtime deps
hydra-core == 1.2.0
pytorch-lightning == 1.7.7
fairscale == 0.4.12

# dev deps
pytest == 7.1.3
black ==  22.10.0
isort == 5.10.1
flake8 == 5.0.4
```

It might still be good to not use * in >= expression though. Personally I think using * in == makes more sense than using it in >=.

**Comment by carmocca** [comment]
Closing as per the above.

The `.*` will be gone anyways once we update to minimum PyTorch 1.10, it's only set for the reasons that Akihiro mentioned.

**Pull request closed without merging**